lib/http-context.js

```diff
@@ -304,6 +306,31 @@ HttpContext.prototype.invoke = function(scope, method, fn, isCtor) {
   var pipe = http && http.pipe;
   var pipeDest = pipe && pipe.dest;
   var pipeSrc = pipe && pipe.source;
+  var streamDesc = method.stream;
+  var methodReturnsStream = !!streamDesc;
+  var res = this.res;
+
+  if (methodReturnsStream) {
+    method.invoke(scope, args, this.options, function(err, results) {
+      if (err) {
+        return fn(err);
+      } else if (streamDesc.json) {
+        res.setHeader('Content-Type',
+          streamDesc.contentType || 'application/json; boundary=NL');
+        res.setHeader('Transfer-Encoding', 'chunked');
+
+        var stream = results[streamDesc.arg];
+
+        if (streamDesc.type === 'ReadableStream') {
+          stream.pipe(JSONStream.stringify(false)).pipe(res);
+        } else {
+          fn(new Error('unsupported stream type'));
+        }
+      } else {
+        fn(new Error('unsupported stream description'));
```

Comment:
Please provide more information to help the user to understand better what they did wrong and how to fix the problem.

Example: `Unsupported stream descriptor, only descriptors with property "json:true" are supported.`.